packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1061,6 +1067,606 @@ describe("SharedTree", () => {
 			unsubscribe();
 		});
 
+		// TODO: try to dedupe some of these tests
+		describe("repair data is destroyed when", () => {
+			it("the collab window progresses far enough", () => {
+				const provider = new TestTreeProviderLite(2);
+				const content = {
+					schema: stringSequenceRootSchema,
+					allowedSchemaModifications: AllowedUpdateType.Initialize,
+					initialTree: ["A", "B", "C", "D"],
+				} satisfies InitializeAndSchematizeConfiguration;
+				const tree1 = schematizeFlexTree(provider.trees[0], content);
+
+				// make sure that revertibles are created
+				const { unsubscribe } = createTestUndoRedoStacks(tree1.checkout.events);
+
+				provider.processMessages();
+				const tree2 = schematizeFlexTree(provider.trees[1], content);
+
+				const root1 = tree1.flexTree;
+				const root2 = tree2.flexTree;
+
+				// get an anchor on the peer to the node we're removing
+				const cursor = tree2.checkout.forest.allocateCursor();
+				moveToDetachedField(tree2.checkout.forest, cursor);
+				cursor.firstNode();
+				assert.equal(cursor.value, "A");
+				const anchor = cursor.buildAnchor();
+				cursor.free();
+
+				// remove in first tree
+				root1.removeAt(0);
+
+				provider.processMessages();
+				const removeSequenceNumber = provider.sequenceNumber;
+				assert.deepEqual([...root1], ["B", "C", "D"]);
+				assert.deepEqual([...root2], ["B", "C", "D"]);
+				assert.equal(tree2.checkout.getRemovedRoots().length, 1);
+
+				// check the detached field on the peer
+				const repairCursor1 = tree2.checkout.forest.allocateCursor();
+				const result1 = tree2.checkout.forest.tryMoveCursorToNode(anchor, repairCursor1);
+				assert.equal(result1, TreeNavigationResult.Ok);
+				assert.equal(repairCursor1.value, "A");
+				repairCursor1.free();
+
+				// send edits to move the collab window up
+				root1.insertAt(0, ["y"]);
+				provider.processMessages();
+				root1.removeAt(0);
+				provider.processMessages();
+
+				assert.deepEqual([...root1], ["B", "C", "D"]);
+				assert.deepEqual([...root2], ["B", "C", "D"]);
+
+				// ensure the remove is out of the collab window
+				assert(removeSequenceNumber < provider.minimumSequenceNumber);
+
+				// check that the repair data on the peer is destroyed
+				const repairCursor2 = tree2.checkout.forest.allocateCursor();
+				const result2 = tree2.checkout.forest.tryMoveCursorToNode(anchor, repairCursor2);
+				assert.equal(result2, TreeNavigationResult.NotFound);
+				repairCursor2.free();
+
+				unsubscribe();
+			});
+
+			it("the collab window progresses far enough - multiple nodes", () => {
```

Comment:
I think it would be sufficient to have this test without the one above (which only removes a single node): it seems really unlikely that the test above would catch a bug that woudn't also be caught by this test.